### DISCUSSION TITLE
Forecast Measures

### DISCUSSION THREAD

**Discussion by JATDALTON** [frePPLe/frepple#665] [Q&A]
I am looking to create a forecast measure to calculate total volume.

In my scenario we are forecasting units, which is working great. We have the volume for each item set and we would like to also calculate the volume.

From the documentation it looks like it should be as simple as 

forecasttotal * item("volume")

However this is causing the demand plan to fail.

Hoping someone can help me with the correct syntax.

**Comment by jdetaeye** [comment]
I just gave it a try, and I think it's broken.

Ticket created #667

**Reply by JATDALTON** [reply]
Thanks, are item distribution attributes supposed to be available for measures as well?

We have different pricing based on item@location/customer and it would be nice to calculate proposed revenue as well.

I can always do this with the API, but thought I would ask.

**Reply by jdetaeye** [reply]
> Thanks, are item distribution attributes supposed to be available for measures as well?

Definitely not.   The forecast editor is all about the **demand** side of the equation.
Item distributions are about the **supply** side of the equation.

Trying to bring supply side info into the forecast editor is a recipe for disaster.

> We have different pricing based on item@location/customer and it would be nice to calculate proposed revenue as well.

You could define a measure of type "local" at the item+location+customer level where you maintain a revenue for that selection. 

We haven't pushed the measures and the expressions very far yet.  So, it's very likely that accessing that measure from an expression will have some limitations as well. It's very low priority for us to extend these capabilities. You're of course more than welcome to submit a pull request.  The majority of the related code is in the files https://github.com/frePPLe/frepple/blob/master/src/forecast/measure_compute.cpp and https://github.com/frePPLe/frepple/blob/master/src/forecast/measure.cpp
